internal/pkg/vm/actor/builtin/account/account.go

```diff
@@ -22,15 +30,21 @@ func NewActor(balance types.AttoFIL) (*actor.Actor, error) {
 	return actor.NewActor(types.AccountActorCodeCid, balance), nil
 }
 
-// UpgradeActor converts the given actor to an account actor, leaving its balance and nonce in place.
```

Comment:
So this is going away?  I never quite understood the security implications but did we change our approach to handle the malicious-pre-creation behavior motivating this?

Reply:
The difference is that we no longer have "Empty" actors. You either send a message to an unknown address and it becomes an Account actor, or you call Exec (usually you call another actor's method that calls Exec) and create an actor of another type. Actors will always have a CID now, though.

Reply:
You probably could, but the miner is actually really going to be added at an ID address. The init actor will then map the new actor address, which will override the mapping of the message previously sent. So it works out in the end. We may want to limit account creation to Secp and BLS addresses.